js/Wikifier.js

```diff
@@ -134,104 +130,102 @@ Wikifier.prototype.subWikifyTerm = function(output,terminatorRegExp)
+		for(var i = 1; i < formatterMatch.length; i++) {
+			if(formatterMatch[i]) {
+				this.formatter.formatters[i - 1].handler(this);
 				this.formatter.formatterRegExp.lastIndex = this.nextMatch;
 				break;
 			}
 		}
 		//# Get the next match
 		terminatorRegExp.lastIndex = this.nextMatch;
 		terminatorMatch = terminatorRegExp.exec(this.source);
-		formatterMatch = this.formatter.formatterRegExp.exec(terminatorMatch ? this.source.substr(0,terminatorMatch.index) : this.source);
+		formatterMatch = this.formatter.formatterRegExp.exec(terminatorMatch ?
+			this.source.substr(0, terminatorMatch.index) : this.source);
 	}
 	//# Output any text after the last match
 	if(this.nextMatch < this.source.length) {
-		this.outputText(this.output,this.nextMatch,this.source.length);
+		this.outputText(this.output, this.nextMatch, this.source.length);
 		this.nextMatch = this.source.length;
 	}
 	//# Restore the output pointer
 	this.output = oldOutput;
 };
 
-Wikifier.prototype.outputText = function(place,startPos,endPos)
+Wikifier.prototype.outputText = function(place, startPos, endPos)
 {
 	//# Check for highlights
-	while(this.highlightMatch && (this.highlightRegExp.lastIndex > startPos) && (this.highlightMatch.index < endPos) && (startPos < endPos)) {
+	while(this.highlightMatch && (this.highlightRegExp.lastIndex > startPos) &&
+		  (this.highlightMatch.index < endPos) && (startPos < endPos)) {
 		//# Deal with any plain text before the highlight
 		if(this.highlightMatch.index > startPos) {
-			createTiddlyText(place,this.source.substring(startPos,this.highlightMatch.index));
+			createTiddlyText(place, this.source.substring(startPos, this.highlightMatch.index));
 			startPos = this.highlightMatch.index;
 		}
 		//# Deal with the highlight
-		var highlightEnd = Math.min(this.highlightRegExp.lastIndex,endPos);
-		createTiddlyElement(place,"span",null,"highlight",this.source.substring(startPos,highlightEnd));
+		var highlightEnd = Math.min(this.highlightRegExp.lastIndex, endPos);
+		createTiddlyElement(place, "span", null, "highlight", this.source.substring(startPos, highlightEnd));
 		startPos = highlightEnd;
 		//# Nudge along to the next highlight if we're done with this one
 		if(startPos >= this.highlightRegExp.lastIndex)
 			this.highlightMatch = this.highlightRegExp.exec(this.source);
 	}
 	//# Do the unhighlighted text left over
 	if(startPos < endPos) {
-		createTiddlyText(place,this.source.substring(startPos,endPos));
+		createTiddlyText(place, this.source.substring(startPos, endPos));
 	}
 };
 
-function wikify(source,output,highlightRegExp,tiddler)
+function wikify(source, output, highlightRegExp, tiddler)
 {
-	if(source) {
-		var wikifier = new Wikifier(source,getParser(tiddler),highlightRegExp,tiddler);
-		var t0 = new Date();
-		wikifier.subWikify(output);
-		if(tiddler && config.options.chkDisplayInstrumentation)
-			displayMessage("wikify:" +tiddler.title+ " in " + (new Date()-t0) + " ms");
-	}
+	if(!source) return;
```

Comment:
well, this is somewhat opinionated, but this is called "bouncer pattern". Basically, reducing nesting usually makes code less complicated, but I don't introduce bouncing in some places where there's 1 or 2 instructions inside `if`

Reply:
... as you wrote it's opinionated. .. Since you are the maintainer now. I can live with that pattern, which I know as "return early"

Reply:
This one is on my list too, it is a time expensive function because of the testing logic and the date assignment in every call.
Please remove the whole testing logic.
If you want some testing use a plugin that provides this.

Reply:
@qbroker this is an interesting point. Could you do some benchmarking to see if this really gives an impact? Does using `if(config.options.chkDisplayInstrumentation) var t0 = new Date();` give the same impact as just removing things? Moving this into a plugin is also an option.

Reply:
@Yakov, I compared different options, no debug logic is the fasted, see attached pdf.
Although the differences are small, this function is called many times. I only stumbled over it when a made a TWC with 50k tiddlers for testing and noticed that it became slow.
[2022-07-wikify-execution-speed.pdf](https://github.com/TiddlyWiki/TiddlyWiki/files/9034861/2022-07-wikify-execution-speed.pdf)

Reply:
@qbroker awesome, great report! 2% seems to be a considerable acceleration for a couple of lines; could you create a PR for this (into `dev` branch)? I suspect there's more room for improvement and also we should create a profiling plugin and move the other bits there, too.